src/shop.js

```diff
@@ -1,5 +1,6 @@
 function BinartaShopjs(checkpoint, deps) {
     var shop = this;
+    var application = deps.application || {profile: function () {}};
```

Comment:
It's interesting you provide a kind of fallback version if no application has been provided but is this really necessary? A null object can be useful in some circumstances but I'm wondering if we don't simply want it to crash if nothing has been provided?